addons/binding/org.openhab.binding.rfxcom/src/main/java/org/openhab/binding/rfxcom/internal/connector/RFXComStreamReader.java

```diff
@@ -37,89 +36,39 @@ public RFXComStreamReader(RFXComBaseConnector connector, InputStream in) {
     }
 
     @Override
-    public void interrupt() {
-        interrupted = true;
-        super.interrupt();
-        try {
-            in.close();
-        } catch (IOException ignore) {
-            // quietly close
-        }
-    }
-
-    @Override
     public void run() {
-        final int dataBufferMaxLen = Byte.MAX_VALUE;
-
-        byte[] dataBuffer = new byte[dataBufferMaxLen];
-
-        int msgLen = 0;
-        int index = 0;
-        boolean startFound = false;
-
         logger.debug("Data listener started");
 
-        try {
-
-            byte[] tmpData = new byte[20];
-            int len = -1;
-
-            while (!interrupted) {
-
-                if ((len = in.read(tmpData)) > 0) {
-
-                    byte[] logData = Arrays.copyOf(tmpData, len);
-                    logger.trace("Received data (len={}): {}", len, DatatypeConverter.printHexBinary(logData));
+        while (!Thread.interrupted()) {
+            try {
+                byte[] buf = new byte[Byte.MAX_VALUE];
 
-                    for (int i = 0; i < len; i++) {
+                if (in.read(buf, 0, 1) > 0 && buf[0] > 0) {
+                    int i = 1;
```

Comment:
i is a index to the buffer. This loop is a pretty standard C idiom however I've increased the verbiage a bit. Personally I think that makes it _harder_ to read but that's because I write a lot more C than Java :-)